lib/pf/UnifiedApi/Controller/Authentication.pm

```diff
@@ -0,0 +1,68 @@
+package pf::UnifiedApi::Controller::Authentication;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Authentication -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Authentication
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller';
+use pf::error qw(is_error);
+use pf::authentication;
+use pf::Authentication::constants qw($LOGIN_SUCCESS);
+
+sub adminAuthentication {
+    my ($self) = @_;
+    
+    my ($status, $json) = $self->parse_json;
+    if (is_error($status)) {
+        $self->render(status => $status, json => $json);
+    }
+
+    my ($result, $roles, $tenant_id) = pf::authentication::adminAuthentication($json->{username}, $json->{password});
+
+    if($result == $LOGIN_SUCCESS) {
+        $self->render(status => 200, json => { result => $result, roles => $roles, tenant_id => ($tenant_id + 0) });
```

Comment:
"+ 0" ?

Reply:
Unfortunately yes, otherwise there is no guarantee that this will be an integer when passed to the JSON library for encoding. Then if its a string containing a number, the JSON decode will not work properly in Golang.

Reply:
You could use the `int` function for clarity.